File: discovery-provider/src/monitors/redis.py
import psutil


def get_redis_num_keys(**kwargs):
    redis = kwargs['redis']
    num_keys = redis.dbsize()
    return num_keys


def get_redis_used_memory(**kwargs):
    redis = kwargs['redis']
    info = redis.info()
    return info['used_memory']


def get_redis_total_memory(**kwargs):
    # The version of redis used does not support total_system_memory
    # redis = kwargs['redis']
    # info = redis.info()
    # return info['total_system_memory']
    mem = psutil.virtual_memory()
    return mem.total

File: creator-node/src/monitors/redis.js
import { redisClient as redis } from '../redis'

/**
 * Parses the result from a redis `info` command
 * @param {string} info
 */
const parseInfo = (info) => {
  const lines = info.split('\r\n')
  const obj = {}
  for (let i = 0, l = info.length; i < l; ++i) {
    let line = lines[i]
    if (line && line.split) {
      line = line.split(':')
      if (line.length > 1) {
        const key = line.shift()
        obj[key] = line.join(':')
      }
    }
  }
  return obj
}

const getRedisNumKeys = async () => {
  const numKeys = await redis.dbsize()
  return numKeys
}

const getRedisUsedMemory = async () => {
  const info = await redis.info()
  const parsedInfo = parseInfo(info)
  return parsedInfo.used_memory
}

const getRedisTotalMemory = async () => {
  const info = await redis.info()
  const parsedInfo = parseInfo(info)
  return parsedInfo.total_system_memory
}

module.exports = {
  getRedisNumKeys,
  getRedisUsedMemory,
  getRedisTotalMemory
}
